tests/test_normalizedCalibrationFlux.py

```diff
@@ -0,0 +1,270 @@
+        """Make a source catalog with option offset and scaling.
+
+        Parameters
+        ----------
+        schema : `lsst.afw.table.Schema`
+            The table schema.
+        num_sources : `int`
+            Number of sources to put into the catalog.
+        ap_flux_offset : `float`
+            Constant offset in aperture fluxes due to "background" issues.
+        cg_scale : `float`
+            Flux ratio between unnormalized flux and aperture (reference) flux.
+        """
+        source_cat = lsst.afw.table.SourceCatalog(schema)
+
+        x = np.random.rand(num_sources)*self.exposure.getWidth() + self.exposure.getX0()
+        y = np.random.rand(num_sources)*self.exposure.getHeight() + self.exposure.getY0()
+        flux = np.random.uniform(low=10000.0, high=100000.0, size=num_sources)
+
+        source_cat.resize(num_sources)
+        source_cat["slot_Centroid_x"] = x
+        source_cat["slot_Centroid_y"] = y
+        source_cat["calib_psf_used"] = True
+
+        # Make a very simple error model.
+        noise_per_pix = 2.0
+        ap_flux_err = np.sqrt((np.pi*12**2.*noise_per_pix)**2. + (flux + ap_flux_offset))
+        cg_flux_err = np.sqrt((np.pi*4**2.*noise_per_pix)**2. + flux)
+
+        source_cat[self.cg_name + "_instFlux"] = cg_scale*flux
+        source_cat[self.cg_name + "_instFluxErr"] = cg_scale*cg_flux_err
+        source_cat[self.cg_name + "_flag"] = np.zeros(num_sources, dtype=bool)
+        source_cat[self.ap_name + "_instFlux"] = flux + ap_flux_offset
+        source_cat[self.ap_name + "_instFluxErr"] = ap_flux_err
+        source_cat[self.ap_name + "_flag"] = np.zeros(num_sources, dtype=bool)
+
+        return source_cat
+
+    def tearDown(self):
+        # del self.schema
+        del self.exposure
+        # del self.norm_task
+
+    def testNormalizedCalibrationFlux(self):
+        np.random.seed(12345)
+        norm_task = self._make_task()
+        catalog = self._make_catalog(norm_task.schema)
+
+        ap_corr_map = norm_task.run(catalog=catalog, exposure=self.exposure).ap_corr_map
+
+        self.assertEqual(
+            catalog.schema.getAliasMap().get("slot_CalibFlux"),
+            norm_task.config.normalized_calibflux_name,
+        )
+
+        self.assertIn("base_CompensatedTophatFlux_12_instFlux", ap_corr_map)
+        self.assertIn("base_CompensatedTophatFlux_12_instFluxErr", ap_corr_map)
+
+        # The full set should have a 1.0 ratio when the
+        # aperture flux offset is 0.0
+        ratio = np.mean(catalog["slot_CalibFlux_instFlux"]/catalog[self.ap_name + "_instFlux"])
+        self.assertFloatsAlmostEqual(ratio, 1.0, rtol=1e-10)
+
+        # The subset that was used should always have a 1.0 ratio.
+        used = catalog["apcorr_base_CompensatedTophatFlux_12_used"]
+        ratio_used = np.mean(
+            catalog["slot_CalibFlux_instFlux"][used]/catalog[self.ap_name + "_instFlux"][used]
+        )
+        self.assertFloatsAlmostEqual(ratio_used, 1.0, rtol=1e-10)
```

Comment:
Is this necessary given the full set check above? It can't hurt, I suppose.

Reply:
Always good to explicitly confirm expectations.